instrumentation/spring/spring-webflux/spring-webflux-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/v5_3/TelemetryProducingWebFilter.java

```diff
@@ -0,0 +1,153 @@
+
+    @Override
+    public void subscribe(CoreSubscriber<? super Void> actual) {
+      if (!instrumenter.shouldStart(parentContext, exchange)) {
+        source.subscribe(actual);
+        return;
+      }
+      Context currentContext = instrumenter.start(parentContext, exchange);
+      try (Scope ignored = currentContext.makeCurrent()) {
+        this.source.subscribe(
+            new TelemetryWrappedSubscriber(actual, currentContext, instrumenter, exchange));
+      }
+    }
+  }
+
+  private static class TelemetryWrappedSubscriber implements CoreSubscriber<Void> {
+
+    private static final HttpServerAttributesGetter<ServerWebExchange, ServerWebExchange>
+        attributesGetter = WebfluxServerHttpAttributesGetter.INSTANCE;
+    private static final SpanNameExtractor<ServerWebExchange> spanNameExtractor =
+        HttpSpanNameExtractor.create(WebfluxServerHttpAttributesGetter.INSTANCE);
+
+    private final CoreSubscriber<? super Void> actual;
+    private final Instrumenter<ServerWebExchange, ServerWebExchange> instrumenter;
+    private final Context currentOtelContext;
+    private final ServerWebExchange exchange;
+
+    TelemetryWrappedSubscriber(
+        CoreSubscriber<? super Void> actual,
+        Context currentOtelContext,
+        Instrumenter<ServerWebExchange, ServerWebExchange> instrumenter,
+        ServerWebExchange exchange) {
+      this.actual = actual;
+      this.instrumenter = instrumenter;
+      this.currentOtelContext = currentOtelContext;
+      this.exchange = exchange;
+    }
+
+    @Override
+    public reactor.util.context.Context currentContext() {
+      return actual.currentContext();
+    }
+
+    @Override
+    public void onSubscribe(Subscription s) {
+      actual.onSubscribe(s);
+    }
+
+    @Override
+    public void onNext(Void unused) {}
+
+    @Override
+    public void onError(Throwable t) {
+      onTerminal(currentOtelContext, t);
+      actual.onError(t);
+    }
+
+    @Override
+    public void onComplete() {
+      onTerminal(currentOtelContext, null);
+      actual.onComplete();
+    }
+
+    private void onTerminal(Context currentContext, Throwable t) {
+      ServerHttpResponse response = exchange.getResponse();
+      if (response.isCommitted()) {
+        end(currentContext, t);
+      } else {
+        response.beforeCommit(
+            () -> {
+              end(currentContext, t);
+              return Mono.empty();
+            });
+      }
+    }
+
+    private void end(Context currentContext, Throwable t) {
+      // Update span name and HTTP route now, because during instrumenter.start()
+      // the HTTP route isn't available from the exchange attributes, but is afterwards
+      String spanName = spanNameExtractor.extract(exchange);
+      String route = attributesGetter.getRoute(exchange);
+      Span span = Span.fromContext(currentContext);
+      span.updateName(spanName);
+      if (route != null) {
+        span.setAttribute(SemanticAttributes.HTTP_ROUTE, route);
```

Comment:
Can you use the `HttpRouteHolder` instead?

Reply:
Done 👍